lib/fizzy/parser.hpp

```diff
@@ -40,8 +40,10 @@ inline parser_result<uint8_t> parse_byte(const uint8_t* pos, const uint8_t* end)
 ///
 /// @param input    The beginning of the expr binary input.
 /// @param end      The end of the binary input.
+/// @param type_idx Index of the type of the function being parsed.
```

Comment:
```suggestion
/// @param type_idx Valid index of the type of the function being parsed.
```

Why not just pass `func_idx` instead?

Reply:
I guess it was handy when the loop in calling `parse_expr` was also validating `type_idx`
It doesn't anymore, and it seems `func_idx` might be more future-proof, so I'll change it.